### ISSUE TITLE
Axum API endpoints should report JSON validation errors

### ISSUE THREAD

**Issue by justinmoon** [fedimint/fedimint#1843]
While prototyping a change to the Core Lightning plugin, I added a new `/htlc` POST endpoint to the Gateway HTTP server which is used to request invoices be paid by clients, and `gateway-cli` commands. 

I was getting a 422 error when posting, and the response didn't say what went wrong. Slowly figured out it had to be JSON validation problem. [This code](https://github.com/fedimint/fedimint/commit/44284c01502174de078327197e80f8c797b84ca6#diff-d9bea38dc9d223b8769cab753343382ddec900194da2cbc982c1a9d519b0ad63R155) includes the validation error in the response. Quickly figured out the problem. Borrowed from [this example](https://github.com/tokio-rs/axum/blob/main/examples/customize-extractor-error/src/with_rejection.rs) in the Axum docs. 

Might be useful to add something like this to all our Axum responses.

**Comment by justinmoon** [comment]
I just noticed this again when updating the gateway UI. It was sending incorrect params to an API call and returns this very vague error:

<img width="464" alt="image" src="https://github.com/fedimint/fedimint/assets/4335621/41a38c8f-f0a5-4ccb-8fc9-3f461bab2675">

**Comment by elsirion** [comment]
@m1sterc001guy is this still relevant?

**Comment by m1sterc001guy** [comment]
I don't think so. Gateway handles errors by logging but is selective in what is returned to clients for privacy reasons. Fine to close I think
